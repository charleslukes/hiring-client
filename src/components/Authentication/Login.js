import React, { useState } from 'react';
import request from '../../request';
import { BeatLoader } from 'react-spinners';
import { Box, Button, Form, Text, FormField } from 'grommet';
import { useSelector, connect } from 'react-redux';
import loginBoundActionCreator from './login.action';
import { errorMessageExtrator } from '../../helpers/utils';

function Login(props) {
  const { error, loading } = useSelector(({ user }) => user);
  const [values, setValues] = useState({
    email: '',
    password: ''
  });
  const { email, password } = values;

  const handleChange = e => {
    const { name, value } = e.target;
    setValues({ ...values, [name]: value });
  };

  const handleSubmit = e => {
    e.preventDefault();
    props.login(values, request);
  };

  return (
    <>
      <Box margin="small" pad="small">
        <Text
          width="auto"
          size="large"
          margin="auto"
          style={{
            fontWeight: 'bold',
            fontSize: '25px',
            paddingBottom: '15px'
          }}
        >
          Login Here
        </Text>
        <Text size="small" alignSelf="center">
          Enter email and password
        </Text>
      </Box>

      {error && (
        <small className="error" style={{ color: 'red', textAlign: 'center' }}>
          {errorMessageExtrator(error)}
        </small>
      )}

      <Form>
        <FormField
          name="email"
          required
          validate={{
            regexp: /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/,
            message: 'Input must be valid email!'
          }}
          value={email}
          onChange={handleChange}
          placeholder="Email"
          type="email"
          style={{
            marginBottom: '15px',
            borderRadius: '20px'
          }}
        />
        <FormField
          placeholder="Password"
          name="password"
          type="password"
          required
          value={password}
          onChange={handleChange}
          // validate={{
          //   regexp: /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
          //   message:
          //     'Password must contain at least 8 characters, 1 letter, and 1 number'
          // }}
          color="dark-1"
          style={{
            marginBottom: '15px',
            borderRadius: '20px'
          }}
        />

        <Box
          direction="row"
          justify="center"
          align="center"
          margin={{ top: 'medium' }}
        >
          <Button
            primary
            width="large"
            color="dark-1"
            label={loading ? <BeatLoader size={5} color="#fff" /> : 'Login'}
            onClick={handleSubmit}
            type="submit"
            style={{ width: '100%', marginTop: 20 }}
          />
        </Box>
      </Form>
    </>
  );
}

const mapDispatchToProps = {
  login: loginBoundActionCreator
};
export default connect(
  null,
  mapDispatchToProps
)(Login);
