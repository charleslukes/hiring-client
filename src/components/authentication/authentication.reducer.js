const initialState = {
  token: ''
};

export default function authentication(state = initialState, action) {
  switch (action.type) {
    case 'SET_TOKEN':
      return {
        ...state,
        token: action.payload
      };
    default:
      return state;
  }
}
