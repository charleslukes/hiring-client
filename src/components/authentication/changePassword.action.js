const changePassBoundActionCreator = (
  data,
  request,
  token,
  activityIndicator,
  handleError,
  onSuccess,
  token
) => async () => {
  try {
    activityIndicator(true);
    const response = await request.put("/users/change-password", data, {
      headers: {
        authorization: `Bearer ${token}`
      }
    });
    onSuccess(response.data);
    activityIndicator(false);
    return response.data;
  } catch (err) {
    handleError(err);
    activityIndicator(false);
  }
};

export default changePassBoundActionCreator;
