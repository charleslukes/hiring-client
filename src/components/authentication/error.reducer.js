import { SIGN_OUT } from './signout.action';
export const initialState = {
  error: ''
};

export default function error(state = initialState, action) {
  switch (action.type) {
    case 'SET_ERROR':
      return {
        ...state,
        error: action.payload
      };
    case SIGN_OUT:
      return {
        ...initialState
      };
    default:
      return state;
  }
}
