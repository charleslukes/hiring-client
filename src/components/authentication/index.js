import React, { lazy } from "react";

import { Route, Switch } from "react-router-dom";
import ProtectedRoute from "../ProtectedRoute";
const Login = lazy(() => import("./Login"));
const SignUp = lazy(() => import("./SignUp"));
const UpdatePassword = lazy(() => import("./UpdatePassword"));
const ChangePassword = lazy(() => import("./ChangePassword"));
const Schedule = lazy(() => import("../dashboard/ScheduleInterview"));
const Invite = lazy(() => import("../InviteHirer"));
const HirerSignUp = lazy(() => import("../HirerSignUp"));
const VerifyHirer = lazy(() => import("../HirerSignUp/VerifyHirer"));
const HirerLogin = lazy(() => import("../HirerSignUp/hirerLogin/HirerLogin"));
const ScheduleInterview = lazy(() => import("../dashboard/ScheduleInterview"));
const InterviewResponse = lazy(() =>
  import("../interviewActivities/InterviewResponse")
);
const ScheduleTest = lazy(() => import("../schedule-test/ScheduleTest"));
const LandingPage = lazy(() => import("../landingPage/"));
export default function(props) {
 
  return (
    <Switch>
      <Route exact path="/" component={LandingPage} />
      <Route exact path="/login" component={Login} />
      <Route exact path="/signup" component={SignUp} />
      <Route exact path="/signup/partner" component={HirerSignUp} />
      <Route exact path="/login/partner" component={HirerLogin} />
      <Route
        exact
        path="/interview-response/:intent/:email/:interviewId"
        component={InterviewResponse}
      />
      <Route
        exact
        path="/update-password/:token/:email"
        component={UpdatePassword}
      />
      <ProtectedRoute>
        <Route
          path="/schedule-interview/:email"
          component={ScheduleInterview}
          
        />
        <Route path="/schedule-test/:email" component={ScheduleTest} />

        <Route exact path="/schedule" component={Schedule} />
        <Route exact path="/invite" component={Invite} />
        <Route
          exact
          path="/verify-hirer/:token/:email"
          component={VerifyHirer}
        />

        <Route
          exact
          path={"/change-password/:token"}
          component={ChangePassword}
        />
      </ProtectedRoute>
    </Switch>
  );
}
