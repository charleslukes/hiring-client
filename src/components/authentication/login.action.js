export const USER_LOGIN = "USER_LOGIN";
const login = payload => ({
  type: USER_LOGIN,
  payload
});

const setToken = payload => ({
  type: "SET_TOKEN",
  payload
});

export const setLoading = payload => ({
  type: "LOADING",
  payload
});
export const onError = payload => ({
  type: "USER_LOGIN_ERROR",
  payload
});

const loginBoundActionCreator = (
  data,
  request,
  navigateToDashboard
) => async dispatch => {
  try {
    dispatch(setLoading(true));
    const response = await request.post("/users/login", data);
    dispatch(login(response.data));
    dispatch(setToken(response.data.token));
    dispatch(setLoading(false));
    navigateToDashboard();
    return response.data;
  } catch (err) {
    dispatch(setLoading(false));
    return dispatch(onError(err));
  }
};

export default loginBoundActionCreator;
