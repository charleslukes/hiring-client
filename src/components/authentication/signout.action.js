export const SIGN_OUT = 'SIGN_OUT';
const signout = () => ({
  type: SIGN_OUT
});

const signOutBoundActionCreator = () => dispatch => {
  dispatch(signout());
};

export default signOutBoundActionCreator;
