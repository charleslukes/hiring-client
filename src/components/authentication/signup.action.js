export const USER_SIGNUP = 'USER_SIGNUP';
const signup = payload => ({
  type: USER_SIGNUP,
  payload
});

const setToken = payload => ({
  type: 'SET_TOKEN',
  payload
});

const setLoading = payload => ({
  type: 'LOADING',
  payload
});
const onError = payload => ({
  type: 'USER_SIGNUP_ERROR',
  payload
});

const signupBoundActionCreator = (data, request) => async dispatch => {
  try {
    dispatch(setLoading(true));
    const response = await request.post('/users/signup', data);
    dispatch(signup(response.data));
    dispatch(setToken(response.data.token));
    dispatch(setLoading(false));
    return response.data;
  } catch (err) {
    dispatch(setLoading(false));
    return dispatch(onError(err));
  }
};

export default signupBoundActionCreator;
