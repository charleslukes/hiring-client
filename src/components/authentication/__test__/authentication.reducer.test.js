import authentication from '../authentication.reducer';

describe('INITIAL_STATE', () => {
  test('is correct', () => {
    const action = { type: 'dummy_action' };
    const initialState = { token: '' };

    expect(authentication(undefined, action)).toEqual(initialState);
  });
});

describe('SET_TOKEN', () => {
  test('returns the correct state', () => {
    const action = { type: 'SET_TOKEN', payload: 'tokenstring' };
    const expectedState = { token: 'tokenstring' };

    expect(authentication(undefined, action)).toEqual(expectedState);
  });
});
