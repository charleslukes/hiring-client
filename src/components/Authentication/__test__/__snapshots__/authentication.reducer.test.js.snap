// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INITIAL_STATE is correct 1`] = `
Object {
  "token": "",
}
`;

exports[`SELECT_AVATAR returns the correct state 1`] = `
Object {
  "token": "tokenstring",
}
`;

exports[`SET_TOKEN returns the correct state 1`] = `
Object {
  "token": "tokenstring",
}
`;
